### ISSUE TITLE
429 Too Many Requests

### ISSUE THREAD

**Issue by Basti189** [iobroker-community-adapters/ioBroker.homeconnect#58]
This entry can be seen in the log file every minute
Start Event Stream Error: {"type":"error","status":429,"message":"Too Many Requests"}

**Comment by TA2k** [comment]
Dann den Adapter mal für 24h pausieren

**Comment by Basti189** [comment]
Ja, das kann man machen. Ich kann aber nicht alle paar Tage den Adapter für 24h ausmachen, ich will den ja eigentlich benutzen :/

**Comment by TA2k** [comment]
Dann startest du den adapter neu oder hast eine Kaffeemaschine die oft aus und an gemacht wird

**Comment by Basti189** [comment]
Ein Neustart ändert nichts and der Situation, ich muss dann trotzdem warten, bis der Adapter wieder Verbindung aufnehmen darf.
Es sind eine Geschirrspühlmaschine und ein Kühlschrank im System, da wird der Status der Türen übertragen, aber ich öffne diese Türen jetzt im Verhältnis nicht sehr oft am Tag

**Comment by TA2k** [comment]
Dann braucht es ein log bevor der 429 Fehler kommt

**Comment by stolly82** [comment]
Hey, ich schließe mich hier mal an. 
Ich lege den Adapter jetzt mal 24h schlafen und werde den auf Debug stellen. Die Logs folgen dann. 

Strange, das hatte ich sicher ein Jahr noch nicht und Geräte wurden nicht geändert. 

Liebe Grüsse

**Comment by stolly82** [comment]
Was mir gerade aufgefallen ist,.... das rate limit von 1000 ist ja ein wenig Niedrig,... oder? 

homeconnect.0 | 2023-04-09 17:21:58.248 | error | Start Event Stream Error: {"type":"error","status":429,"message":"Too Many Requests"}
-- | -- | -- | --
homeconnect.0 | 2023-04-09 17:21:58.246 | error | 429 Too Many Requests
homeconnect.0 | 2023-04-09 17:21:58.043 | error | {"error":{"description":"The rate limit \"1000 calls in 1 day\" was reached. Requests are blocked during the remaining period of 34354 seconds.","key":"429"}}
homeconnect.0 | 2023-04-09 17:21:58.041 | error | AxiosError: Request failed with status code 429
homeconnect.0 | 2023-04-09 17:21:57.567 | info | starting. Version 1.1.1 in /opt/iobroker/node_modules/iobroker.homeconnect, node: v16.19.1, js-controller: 4.0.24

**Comment by Basti189** [comment]
Ich hab den Adapter heute das erste mal wieder seit Wochen in Betrieb genommen und habe jetzt auch die Meldung, dass das Limit von 1000 Anfragen pro Tag erreicht ist. Zum Steuern reicht das ganze eigentlich, aber um auf Ereignisse an den Geräte zu reagieren, sind die 1000 ja echt bescheiden, würde ja eine Abfrage nur alle 1,44 Minuten ermöglichen :/

**Comment by TA2k** [comment]
Ereignis sind keine Anfragen. Die 1000 beziehen sich nur auf die initialierung und befehle die gesendet werden

**Comment by andispammonyer** [comment]
Hab dasselbe Problem. Ich lasse den Adapter alle 6 Std. neustarten um eventuelle Hänger zu vermeiden.
Alle 6 Std. würde ja auch nur bedeuten es gibt 4x Requests von den besagten 1000 pro Tag. Richtig?
Ich habe hierzu die Einstellungen so gesetzt: * */6 * * *

**Comment by Bernd9000** [comment]
Hier auch, neue Waschmaschine, bisschen getestet, nach zwei Stunden sind angeblich die 1000 erreicht.
Ich glaube da wird alles mögliche mitgezählt.

In dem Adapter sollte zum Test mal ein Zähler eingebaut werden wieviel Anfragen tatsächlich stattgefunden haben und das dann ab die die Log.

**Comment by moba-tt** [comment]
Habe das gleiche Problem im iobroker, Neustarten des Adapter hilft nicht.

**Comment by TA2k** [comment]
Ja wie es im Log steht einfach 24h warten bis das Limit wieder aufgehoben ist.

**Comment by Geophysiker** [comment]
Moin, Moin

Ich habe den Adapter jetzt schon Jahren "im Gebrauch" und warte immer noch das er endlich funtioniert.
Ich schalte ihn ein und nach einigen Stunden wegen "429 Too Many Requests" aus. Ich weiss einfach nicht was ich machen soll. Ich habe Waschmaschine, Trockner, Geschirrspühler, Backofen und Herdplatten, scheint wohl zu viel zu sein. Bitte .... 

Tschö.

**Comment by TA2k** [comment]
Bei sovielen geraten kann der Adapter nur einmal am Tag gestartet werden einfach ein paar Tage laufen lassen bis er sich einpegelt

Oder "Deaktiviere Programme holen bei bei  Verbindungsaufbau (reduziert rate limit)" wen die geräte oft ausgeschaltet werden

**Comment by Geophysiker** [comment]
Moin, Moin

Geräte werden normal genutzt, also kein ständiges Ein und Ausschalten.

Habe den Adapter um 19:26h gestartet und um 00:26h:
----------------
Too many requests. Please wait 24h.
Start Event Stream Error: {"type":"error","status":429,"message":"Too Many Requests"}
429 Too Many Requests
-----------------

Ist also zu nichts zu gebrauchen. :((

Tschö

**Comment by TA2k** [comment]
Im debug log sieht man genau welches Geräte dauernd ausgeschaltet wird
bzw die Option
Deaktiviere Programme holen bei bei Verbindungsaufbau (reduziert rate limit)
nutzen

**Comment by Geophysiker** [comment]
Moin, Moin 

Sorry aber wie ich das rate limit ändere weiss ich leider nicht.:(

Hatte das debug log laufen
----------------
2025-01-13 22:36:53.904  - [34mdebug[39m: homeconnect.0 (346245) [object Object]
2025-01-13 22:37:48.905  - [34mdebug[39m: homeconnect.0 (346245) [object Object]
2025-01-13 22:38:43.908  - [34mdebug[39m: homeconnect.0 (346245) [object Object]
2025-01-13 22:39:38.909  - [34mdebug[39m: homeconnect.0 (346245) [object Object]
2025-01-13 22:40:33.908  - [34mdebug[39m: homeconnect.0 (346245) [object Object]
2025-01-13 22:40:40.481  - [34mdebug[39m: homeconnect.0 (346245) [object Object]
2025-01-13 22:41:28.906  - [34mdebug[39m: homeconnect.0 (346245) [object Object]
usw.
---------------------
Da steht nichts anderes.

Tschö

**Comment by andispammonyer** [comment]
nein bei mir nicht mehr

**Comment by Shakira10** [comment]
Ja, habe immer noch das selbe Problem...

**Comment by Geophysiker** [comment]
Ich  nutze jetzt Homeconnect Cloudless

**Comment by Shakira10** [comment]
@Geophysiker 
Homeconnect Cloudless Adapter kann ich nicht finden? 
Danke 
Shakira

**Comment by mcm1957** [comment]
Einen Adapter homeconnect-cloudlrss kenne ich nicht. Wahrscheinlich ist der Adapter cloudlrss-homeconnect gemeint der derzeit nur als BETA Testversion verfügbar ist.

Bitte aber Diskussionen zu anderen Adaptern im Forum führenbsofern sie nicht eine Eingrenzung / Behebung des in diesem Topic erfasstenbProblems darstellen.
